x/gamm/simulation/operations.go

```diff
@@ -135,3 +163,92 @@ func SimulateMsgCreatePool(ak stakingTypes.AccountKeeper, bk stakingTypes.BankKe
 			r, app, txGen, &msg, spentCoins, ctx, simAccount, ak, bk, types.ModuleName)
 	}
 }
+
+// SimulateMsgSwapExactAmountIn generates a MsgSwapExactAmountIn with random values
+func SimulateMsgSwapExactAmountIn(ak stakingTypes.AccountKeeper, bk stakingTypes.BankKeeper, k keeper.Keeper) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		simAccount, _ := simtypes.RandomAcc(r, accs)
+		simCoins := bk.SpendableCoins(ctx, simAccount.Address)
+		if simCoins.Len() <= 0 {
+			return simtypes.NoOpMsg(
+				types.ModuleName, types.TypeMsgSwapExactAmountIn, "Account have no coin"), nil, nil
+		}
+
+		coin := simCoins[r.Intn(len(simCoins))]
+		amt, _ := simtypes.RandPositiveInt(r, coin.Amount)
+
+		tokenIn := sdk.Coin{
+			Denom: coin.Denom,
+			Amount: amt,
+		}
+
+		routes, tokenOut := RandomExactAmountInRoute(ctx, r, k, tokenIn)
+		if routes == nil {
+			return simtypes.NoOpMsg(
+				types.ModuleName, types.TypeMsgSwapExactAmountIn, "No pool exist"), nil, nil
+		}
+
+		tokenOutMin, _ := simtypes.RandPositiveInt(r, tokenOut.Amount)
+
+		msg := types.MsgSwapExactAmountIn{
+			Sender:             simAccount.Address.String(),
+			Routes: routes,
+			TokenIn: tokenIn,
+			TokenOutMinAmount: tokenOutMin,
+		}
+
+		txGen := simappparams.MakeTestEncodingConfig().TxConfig
+		return osmo_simulation.GenAndDeliverTxWithRandFees(
+			r, app, txGen, &msg, sdk.Coins{tokenIn}, ctx, simAccount, ak, bk, types.ModuleName)
+	}
+}
+
+func RandomExactAmountInRoute(ctx sdk.Context, r *rand.Rand, k keeper.Keeper, tokenIn sdk.Coin) (res []types.SwapAmountInRoute, tokenOut sdk.Coin) {
+	routeLen := r.Intn(1)+1
+
+	pools, err := k.GetPools(ctx)
```

Comment:
I'm a bit worried this will slow down the simulator significantly. Lets leave it for now, and check back on this in the future